compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1283,8 +1283,8 @@ object Parsers {
           in.token == IDENTIFIER && in.name == nme.extension
         case PackageDef(pid: RefTree, _) =>
           in.isIdent && in.name == pid.name
-        case PatDef(_, IdPattern(id, _) :: Nil, _, _) =>
-         in.isIdent && in.name == id.name
+        case PatDef(_, IdPattern(id, _) :: Nil, _, _) => // TODO: it looks like this case is never reached
```

Comment:
What is `PatDef` responsible for, something like:
`val (a, b) = (1, 3)`
?

Reply:
Ok, looks like that based on https://dotty.epfl.ch/docs/internals/syntax.html#declarations-and-definitions

However, when working on the tree in Metals I think we never get PatDef, but `ValDef($1$, TypeTree[/....`
Could it be connected?

Reply:
Also, maybe related to https://github.com/lampepfl/dotty/issues/11919 ?

Reply:
looking at `IdPattern` doc I am thinking this is what `val x: Int = 3` looks like, I put `???` on this branch but could never trigger it.

Reply:
it looks like it is escaped already in `patDefOrDcl` in Parsers

Reply:
Can you add a comment removing this case then?

Reply:
err, commit

Reply:
done